src/services/api-service.tsx

```diff
@@ -0,0 +1,100 @@
+import { IProvider } from "../interfaces/iprovider";
+import { Observable } from "rxjs";
+import { Weather } from "../models/Weather";
+import { ApiInfo } from './api-info';
+import { Coordinate } from "../models/Coordinate";
+import { HourlyForecast } from "../models/HourlyForecast";
+/*
+  The purpose of this class is to be the main point of API consume.
+  This class will be used as the Service that is allowed to request using REST to OpenWeather API
+ */
+export class ApiService {
+
+    ApiInfo: ApiInfo = new ApiInfo();
+
+    constructor(private provider: IProvider) { }
+  
+    /**
+     * This method returns the weather for the given city
+     * @param cityName Name of the city
+     */
+    getCurrentWeather(cityName: string): Observable<Weather> {
+      return Observable.create((observer: { next: (arg0: any) => void; }) => {
+        this.provider.get(`/weather?q=${cityName}&appid=${this.ApiInfo.API_KEY}`).subscribe(
```

Comment:
Is this code working? The constructor is not assigning any reference to the `provider` attribute.

Reply:
The idea is that an instance of HttpService will be injected on the provider attribute since HttpService is implementing IProvider interface.